libnetwork/drivers/bridge/setup_ip_tables_linux_test.go

```diff
@@ -464,27 +467,7 @@ func TestMirroredWSL2Workaround(t *testing.T) {
 	} {
 		t.Run(tc.desc, func(t *testing.T) {
 			defer netnsutils.SetupTestOSContext(t)()
-
-			if tc.loopback0 {
-				loopback0 := &netlink.Dummy{
-					LinkAttrs: netlink.LinkAttrs{
-						Name: "loopback0",
-					},
-				}
-				err := netlink.LinkAdd(loopback0)
-				assert.NilError(t, err)
-			}
-
-			if tc.wslinfoPerm != 0 {
-				wslinfoPathOrig := wslinfoPath
-				defer func() {
-					wslinfoPath = wslinfoPathOrig
-				}()
-				tmpdir := t.TempDir()
-				wslinfoPath = filepath.Join(tmpdir, "wslinfo")
-				err := os.WriteFile(wslinfoPath, []byte("#!/bin/sh\necho dummy file\n"), tc.wslinfoPerm)
-				assert.NilError(t, err)
-			}
+			defer setupWSL2MirroredMode(t, tc.loopback0, tc.wslinfoPerm)()
```

Comment:
Really a nit as well; wondering if it's more descriptive if an intermediate var would be used here;

```go
restoreWSLinfoPath := setupWSL2MirroredMode(t, tc.loopback0, tc.wslinfoPerm)
defer restoreWSLinfoPath()
```